src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -233,6 +252,10 @@ object WdlExpression {
         val lhs = toString(a.getAttribute("lhs"), highlighter)
         val rhs = toString(a.getAttribute("rhs"), highlighter)
         s"$lhs.$rhs"
+      case a: Ast if a.isArrayOrMapIndex =>
+        val lhs = toString(a.getAttribute("lhs"), highlighter)
+        val rhs = toString(a.getAttribute("rhs"), highlighter)
+        s"$lhs[$rhs]"
     }
```

Comment:
I noticed that too.  I'll see if I can fill in the blanks.  This function grew organically, we only added what we needed.